code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -148,26 +148,34 @@
 /datum/reagent/drug/methamphetamine
 	name = "Methamphetamine"
 	id = "methamphetamine"
-	description = "Reduces stun times by about 300%, speeds the user up, and allows the user to quickly recover stamina while dealing a small amount of Brain damage. If overdosed the subject will move randomly, laugh randomly, drop items and suffer from Toxin and Brain damage. If addicted the subject will constantly jitter and drool, before becoming dizzy and losing motor control and eventually suffer heavy toxin damage."
+	description = "Reduces stun times by about 300% and allows the user to quickly recover stamina while dealing a small amount of Brain damage. Breaks down slowly into histamine and hits the user with a large amount of histamine if they are stunned. Reacts badly with Ephedrine. If overdosed the subject will move randomly, laugh randomly, drop items and suffer from Toxin and Brain damage. If addicted the subject will constantly jitter and drool, before becoming dizzy and losing motor control and eventually suffer heavy toxin damage."
 	reagent_state = LIQUID
 	color = "#FAFAFA"
 	overdose_threshold = 20
 	addiction_threshold = 10
-	metabolization_rate = 0.75 * REAGENTS_METABOLISM
 
 /datum/reagent/drug/methamphetamine/on_mob_life(mob/living/M)
-	var/high_message = pick("You feel hyper.", "You feel like you need to go faster.", "You feel like you can run the world.")
+	var/high_message = pick("You feel hyper!", "You feel like you're unstoppable!", "You feel like you can take on the world!")
 	if(prob(5))
 		M << "<span class='notice'>[high_message]</span>"
 	M.AdjustParalysis(-2, 0)
 	M.AdjustStunned(-2, 0)
 	M.AdjustWeakened(-2, 0)
 	M.adjustStaminaLoss(-2, 0)
-	M.status_flags |= GOTTAGOREALLYFAST
 	M.Jitter(2)
 	M.adjustBrainLoss(0.25)
 	if(prob(5))
 		M.emote(pick("twitch", "shiver"))
+		M.reagents.add_reagent("histamine", rand(2,6))
+	if(M.stunned || M.weakened)//If you get stunned you do not get off scott free
+		if(prob(50))
+			M.reagents.add_reagent("histamine", 10)
+			M.adjustToxLoss(rand(5,15), 0)
+	if(M.reagents.has_reagent("diphenhydramine"))
+		if(prob(20))
+			M << "<span class='boldwarning'>Mixing diphenhydramine and Meth makes your stomach, head and everything fucking spin!</span>"
+			M.reagents.add_reagent("skewium", rand(1,5))
```

Comment:
it has to mix IN a mob so i think the effort applied resonable. 

Plus making meth and Diphendramine together his about the same amount of effort.